src/rnn/mikolov_rnn.cc

```diff
@@ -177,29 +177,33 @@ Status MikolovModelReader::parse() {
   auto maxBlock = std::max<u64>({hdr.layerSize * hdr.vocabSize, hdr.maxentSize,
                                  hdr.layerSize * hdr.layerSize});
   // 3 comes from rounding to next value + sizeof(float)
-  auto pageSize = 1ULL << (static_cast<u32>(std::log2(maxBlock)) + 3);
+  auto pageSize = static_cast<size_t>(1) << (static_cast<u32>(std::log2(maxBlock)) + 3);
   data_->memmgr.initialize(pageSize);
   data_->alloc = data_->memmgr.value().core();
   auto& alloc = data_->alloc;
+
+  const auto embedding_size = static_cast<size_t>(hdr.layerSize) * static_cast<size_t>(hdr.vocabSize);
```

Comment:
Could you name this embedding_matrix_size
Usually an embedding for a single token is called embedding.

Next thing should be called transition_matrix_size in that case.

Reply:
Renamed it

Reply:
transition_matrix_size as well please!

Anyway, congratulations on 1000th commit :p

Reply:
Oh sorry, I thought you meant it as if we leave name as it is

Reply:
Sorry for the confusion.

Need to improve my clearness.

Reply:
Addressed.

And thanks, sorry about ruining nice number `999` 😄